net/netns/netns_darwin.go

```diff
@@ -33,60 +35,154 @@ var bindToInterfaceByRouteEnv = envknob.RegisterBool("TS_BIND_TO_INTERFACE_BY_RO
 
 var errInterfaceStateInvalid = errors.New("interface state invalid")
 
-// controlLogf marks c as necessary to dial in a separate network namespace.
-//
-// It's intentionally the same signature as net.Dialer.Control
-// and net.ListenConfig.Control.
+// routeCache caches the results of interfaceIndexFor calls to avoid
+// spamming the AF_ROUTE socket. This is used for soft
+// isolation mode where we do many route lookups.
+type routeCacheEntry struct {
+	ifIndex int
+	err     error
+}
```

Comment:
Instead of caching error results you should just not add them to the cache so they can be looked up every time. The cache should just be `[string, int]`